beets/dbcore/query.py

```diff
@@ -428,6 +416,28 @@ def col_clause(self) -> Tuple[str, Sequence[SQLiteType]]:
                 return "1", ()
 
 
+class InQuery(FieldQuery[Sequence[AnySQLiteType]]):
```

Comment:
```suggestion
class InQuery(FieldQuery[Sequence[Generic[AnySQLiteType]]]):
```

Making `InQuery` generic over the pattern type would allow subclasses to narrow this down (e.g. to `BLOB_TYPE a.k.a. memoryview` for `PlayListQuery`). (I didn't test this change, but I think this should work.)

Reply:
Completely! I started thinking about this but I couldn't see a quick and straightforward way of implementing this. Let me have a look what can be done here

Reply:
I'm not totally sure, but I think what I suggested above might already be enough, since `AnySQLiteType` is a `TypeVar` with an appropriate bound. Then, it would simply become `class PlayListQuery(InQuery[BLOB_TYPE])`.

Reply:
Have tried it out but mypy has different plans 😅 
![image](https://github.com/beetbox/beets/assets/16212750/b2bae9ae-ce52-429e-9600-24ce18448404)

Reply:
Oh, I think it needs to be
```py
class InQuery(Generic[AnySQLiteType], FieldQuery[Sequence[AnySQLiteType]]):
```
that seems to type-check for me locally, and in hindsight makes much more sense: `InQuery` is generic over `AnySQLiteType`, and then we immediately use that type variable again for naming the parent class.

Reply:
Thanks! This works fine now, see the latest commits